File: typescript/arreglos.ts
export function crearAleatorio(n = 10): number[] { return Array.from({ length: n }, () => Math.floor(Math.random() * 100) + 1); }
export function recorrerForClasico(a: number[]): void { for (let i = 0; i < a.length; i++) console.log(`a[${i}] = ${a[i]}`); }
export function recorrerForEach(a: number[]): void { a.forEach(valor => process.stdout.write(`${valor} `)); console.log(); }
export function imparesACero(a: number[]): void { for (let i = 0; i < a.length; i++) if (a[i] % 2 !== 0) a[i] = 0; }
export function multiplicarPorIndice(a: number[]): void { for (let i = 0; i < a.length; i++) a[i] *= i; }
export function busquedaLineal(a: number[], objetivo: number): number { for (let i = 0; i < a.length; i++) if (a[i] === objetivo) return i; return -1; }
export function demostrarArreglos(): void { const original = crearAleatorio(); console.log('Original:', original); recorrerForClasico(original); recorrerForEach(original); const sinImpares = [...original]; imparesACero(sinImpares); console.log('Impares a cero:', sinImpares); const porIndice = [...original]; multiplicarPorIndice(porIndice); console.log('Por indice:', porIndice); console.log('Busqueda:', busquedaLineal(original, original[3])); }

File: csharp/Arreglos.cs
namespace ProtocoloColaborativo;

public static class Arreglos
{
    private static readonly Random Random = new();
    public static int[] CrearAleatorio(int n) => Enumerable.Range(0, n).Select(_ => Random.Next(1, 101)).ToArray();
    public static void RecorrerForClasico(int[] arreglo) { for (int i = 0; i < arreglo.Length; i++) Console.WriteLine($"a[{i}] = {arreglo[i]}"); }
    public static void RecorrerForEach(int[] arreglo) { foreach (int valor in arreglo) Console.Write($"{valor} "); Console.WriteLine(); }
    public static void ImparesACero(int[] arreglo) { for (int i = 0; i < arreglo.Length; i++) if (arreglo[i] % 2 != 0) arreglo[i] = 0; }
    public static void MultiplicarPorIndice(int[] arreglo) { for (int i = 0; i < arreglo.Length; i++) arreglo[i] *= i; }
    public static int BusquedaLineal(int[] arreglo, int objetivo) { for (int i = 0; i < arreglo.Length; i++) if (arreglo[i] == objetivo) return i; return -1; }
    public static void Demostrar()
    {
        int[] original = CrearAleatorio(10);
        Console.WriteLine("Arreglo original: " + string.Join(", ", original));
        RecorrerForClasico(original); RecorrerForEach(original);
        int[] sinImpares = (int[])original.Clone(); ImparesACero(sinImpares); Console.WriteLine("Impares a cero: " + string.Join(", ", sinImpares));
        int[] porIndice = (int[])original.Clone(); MultiplicarPorIndice(porIndice); Console.WriteLine("Multiplicado por indice: " + string.Join(", ", porIndice));
        Console.WriteLine($"Busqueda lineal: {BusquedaLineal(original, original[3])}");
    }
}
